src/compiler/checker.ts

```diff
@@ -21106,7 +21106,15 @@ namespace ts {
             }
 
             if (isPartOfTypeNode(node)) {
-                return getTypeFromTypeNode(<TypeNode>node);
+                let typeFromTypeNode = getTypeFromTypeNode(<TypeNode>node);
+
+                if (typeFromTypeNode && isExpressionWithTypeArgumentsInClassImplementsClause(node)) {

```

Comment:
Is this the right thing to check for to find out if we are in an `implements` clause?